src/utils/IPAddress.ts

```diff
@@ -1,10 +1,16 @@
 import type { IPAddress } from "../Types/strings";
-import { getRandomByte } from "./helpers/getRandomByte";
 
 /**
  * Generate a random IP address
  * Does not check to see if the IP already exists in the game
  */
 export const createRandomIp = (): IPAddress => {
-  return `${getRandomByte(99)}.${getRandomByte(9)}.${getRandomByte(9)}.${getRandomByte(9)}` as IPAddress;
+  // Credit goes to yichizhng on BitBurner discord
+  // Generates a number like 0.c8f0a07f1d47e8
+  const ip = Math.random().toString(16);
+  // uses regex to match every 2 characters. [0.][c8][f0][a0][7f][1d][47][e8]
+  // we only want #1 through #4
+  const sliced = ip.match(/../g).slice(1, 5);
```

Comment:
```suggestion
  const matchResult = ip.match(/../g);
  if (!matchResult) {
    // This case should never happen.
    throw new Error(`Unexpected regex matching bug in createRandomIp. ip: ${ip}`);
  }
  const sliced = matchResult.slice(1, 5);
```
`String.match` may return null. In this case, it's not a problem because `Math.random().toString(16)` always matches `/../g`, unless `Math.random` is overridden. Nonetheless, we should always check the null case.

I'm not sure why our lint setup (`npm run lint`) does not catch this case, but the lint setup in my IDE catches it.